db/sql/sql_test.go

```diff
@@ -422,6 +468,35 @@ var _ = Describe("Sql", func() {
 				checkAndStatement(resSql, "`test_string` = ?", "`test_number` = ?", param, []interface{}{42, "123"})
 			})
 		})
+		checkLikeDisjunctionAndConjunction := func(searchString string, numberString string, filterType string, convertedString string, convertedNumber string){
```

Comment:
I think we can consider one step more for generalized helper function and readable test cases not only for your added test cases. Does following function definition work for you?

```
checkTranslationFilterToWhereClause := func(filter map[string]interface{}, whereClause squirrel.Sqlizer) {
	res, err := AddFilterToQuery(testSchema, query, filter, false)

	Expect(err).ToNot(HaveOccurred())
	resSql, param, err := res.ToSql()
	Expect(err).ToNot(HaveOccurred())

	expectedQuery = expectedQuery.Where(whereClause)
	expectedSql, expectedParam, _ := expectedQuery.ToSql()
	Expect(resSql).To(Equal(expectedSql))
	Expect(param).To(Equal(expectedParam))
}
```

Maybe above helper function can be used by not only "Disjunction and conjunction" context but also "Basic select query".